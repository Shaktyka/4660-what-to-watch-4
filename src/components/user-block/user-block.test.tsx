import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import UserBlock from './user-block';

describe(`UserBlock rendering`, () => {

  it(`UserBlock renders correctly`, () => {
    const userData = {
      id: 1,
      avatar: `/img.jpg`,
      name: `Igar`,
      email: `t@li.ru`
    };

    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              isAuthorized={true}
              userData={userData}
            />
          </BrowserRouter>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
