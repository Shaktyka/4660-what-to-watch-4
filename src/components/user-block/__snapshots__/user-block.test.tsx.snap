// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock rendering UserBlock renders correctly 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__avatar"
    href="/mylist"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <img
      alt="Igar's avatar"
      height="63"
      src="https://4.react.pages.academy/img.jpg"
      width="63"
    />
  </a>
</div>
`;
